java/src/jmri/jmrit/beantable/BlockTableAction.java

```diff
@@ -141,6 +144,83 @@ public void setMenuBar(BeanTableFrame<Block> f) {
+        jrbmi.addItemListener(new ItemListener() {
+            @Override
+            public void itemStateChanged(ItemEvent e) {
+                setRestoreRule(RestoreRule.RESTOREALWAYS);
+            }
+        });
+        valuesButtonGroup.add(jrbmi);
+        valuesMenu.add(jrbmi);
+        jrbmi.setSelected(_restoreRule == RestoreRule.RESTOREALWAYS);
+
+        jrbmi = new JRadioButtonMenuItem(Bundle.getMessage("ValuesMenuRestoreOccupiedOnly"));  // NOI18N
+        jrbmi.addItemListener(new ItemListener() {
+            @Override
+            public void itemStateChanged(ItemEvent e) {
+                setRestoreRule(RestoreRule.RESTOREOCCUPIEDONLY);
+            }
+        });
+        valuesButtonGroup.add(jrbmi);
+        valuesMenu.add(jrbmi);
+        jrbmi.setSelected(_restoreRule == RestoreRule.RESTOREOCCUPIEDONLY);
+
+        jrbmi = new JRadioButtonMenuItem(Bundle.getMessage("ValuesMenuRestoreOnlyIfAllOccupied"));  // NOI18N
+        jrbmi.addItemListener(new ItemListener() {
+            @Override
+            public void itemStateChanged(ItemEvent e) {
+                setRestoreRule(RestoreRule.RESTOREONLYIFALLOCCUPIED);
+            }
+        });
+        valuesButtonGroup.add(jrbmi);
+        valuesMenu.add(jrbmi);
+        jrbmi.setSelected(_restoreRule == RestoreRule.RESTOREONLYIFALLOCCUPIED);
+
+        menuBar.add(valuesMenu, pos + offset + 2); // put it to the right of the Speed menu
+    
+    }
+
+    /**
+     * Save the restore rule selection. Called by menu item change events.
+     *
+     * @param newRule The RestoreRule enum constant
+     */
+    void setRestoreRule(RestoreRule newRule) {
+        _restoreRule = newRule;
+        InstanceManager.getDefault(jmri.UserPreferencesManager.class).
+                setProperty(getClassName(), "Restore Rule", newRule.toString());  // NOI18N
+    }
+    
+    /**
+     * Retrieve the restore rule selection from user preferences
+     *
+     * @return restoreRule 
+     */
+    public static RestoreRule getRestoreRule() {
+        RestoreRule rr = RestoreRule.RESTOREONLYIFALLOCCUPIED; //default to previous JMRI behavior 
+        Object rro = InstanceManager.getDefault(jmri.UserPreferencesManager.class).
+                getProperty("jmri.jmrit.beantable.BlockTableAction", "Restore Rule");      // NOI18N
+        if (rro != null) {
+            String cr = (String) rro;
+            switch (cr) {
+                case "RESTOREALWAYS":        // NOI18N
+                    rr = RestoreRule.RESTOREALWAYS;
+                    break;
+                case "RESTOREOCCUPIEDONLY":       // NOI18N
+                    rr = RestoreRule.RESTOREOCCUPIEDONLY;
+                    break;
+                case "RESTOREONLYIFALLOCCUPIED":        // NOI18N
+                    rr = RestoreRule.RESTOREONLYIFALLOCCUPIED;
+                    break;
+                default:
+                    log.warn("Invalid Block Restore Rule value '{}' ignored", cr);  // NOI18N
+            }
```

Comment:
No need to change, but you can use the method `RestoreRule.valueOf(String)` to get the RestoreRule value.
https://docs.oracle.com/javase/8/docs/api/java/lang/Enum.html#valueOf-java.lang.Class-java.lang.String-

The benefit of that is if someone later adds a new enum to RestoreRule. The `valueOf()` will then also handle that new value.